angstudy/src/app/components/star-rating/star-rating.component.ts

```diff
@@ -1,61 +1,27 @@
-import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
+import { stars } from './constants/stars'; 
 
 @Component({
-  selector: 'app-star-rating',
-  templateUrl: './star-rating.component.html',
-  styleUrls: ['./star-rating.component.sass']
+	selector: 'app-star-rating',
+	templateUrl: './star-rating.component.html',
+	styleUrls: ['./star-rating.component.css'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class StarRatingComponent implements OnInit {
-  selectedRating = 0;
+export class StarRatingComponent {
+	readonly stars = stars;
 
-  @Input() totalFeedBack: number = 0;
+	@Input() public feedbacksCount?: number = 0;
+	@Input() public rate?: number = -1;
+	@Input() public readOnly: boolean = false;
 
-  stars = [
-    {
-      id: 1,
-      icon: 'star',
-      class: 'star-gray star-hover star'
-    },
-    {
-      id: 2,
-      icon: 'star',
-      class: 'star-gray star-hover star'
-    },
-    {
-      id: 3,
-      icon: 'star',
-      class: 'star-gray star-hover star'
-    },
-    {
-      id: 4,
-      icon: 'star',
-      class: 'star-gray star-hover star'
-    },
-    {
-      id: 5,
-      icon: 'star',
-      class: 'star-gray star-hover star'
-    }
-  ];
+	constructor() {}
 
-  constructor() {}
-  ngOnInit(): void {
-  }
-
-  selectStar(value: number): void{
-    // prevent multiple selection
-    this.stars.filter( (star) => {
-      if ( star.id <= value){
-        star.class = 'star-gold star';
-      }else{
-        star.class = 'star-gray star';
-      }
-      return star;
-    });
-
-    if (this.selectedRating === 0) {
-      this.totalFeedBack++;
-    }
-    this.selectedRating = value;    
-  }
-}
\ No newline at end of file
+	selectStar(index: number): void {
+		if (!this.readOnly){
```

Comment:
Большая вложенность, давай использовать ранние return
```suggestion
		if (this.readOnly) {
		     return;
		}
		
		if (this.rate === -1) {
			this.feedbacksCount ? this.feedbacksCount++ : 1; // эту строчку не понял, для чего она, возможно стоит заменить на this.feedbacksCount += 1;
		}
		
		this.rate = index;
```

Reply:
this.feedbacksCount может быть undefined